shared/src/main/scala/effekt/Typer.scala

```diff
@@ -561,10 +573,18 @@ class Typer extends Phase[Module, Module] { typer =>
       val (tpe2 / stmtEffs) = arg.body checkAgainst tpe1
 
       subst = subst union Substitution.unify(tpe1, tpe2)
-      effs = (effs ++ (stmtEffs -- handled))
+      if (!isResume) {
+        effs = (effs ++ (stmtEffs -- handled))
+      }
     }
 
-    (params zip args) foreach { case (ps, as) => checkArgumentSection(ps, as) }
+    (params zip args) foreach {
+      // Special case - treat 1st param type () => T as T
+      case (List(bt: BlockType), arg @ source.ValueArgs(_)) if sym.name.name == "resume" => {
```

Comment:
Again here, checking whether the symbol is an instance of

https://github.com/long-long-float/effekt/blob/7020b8314268437b5af00fc403eee05771b9dd06/shared/src/main/scala/effekt/symbols/symbols.scala#L90

is more reliable.